### PR TITLE
[cluster_test] Add --duration to --health-check

### PR DESCRIPTION
**Pull request by changyonghu**
Closes: #4066

### CODE REVIEW

File: testsuite/cluster-test/src/main.rs [hunk 1]
```diff
@@ -738,16 +742,24 @@ impl ClusterTestRunner {
...
             let events = self.logs.recv_all_until_deadline(deadline);
-            let _ignore =
-                self.health_check_runner
-                    .run(&events, &HashSet::new(), PrintFailures::All);
+            let result = self
+                .health_check_runner
+                .run(&events, &HashSet::new(), PrintFailures::All);
+            let now = Instant::now();
+            if now > health_check_deadline {
+                break match result {
```
Comment (ankushagarwal) [inline testsuite/cluster-test/src/main.rs:758]: You can simplify this to `break result.map(|_| ())`

**Comment by ankushagarwal** [comment]
@changyonghu : Were you able to test that this works.

LGTM other than minor comment

**Comment by changyonghu** [comment]
> @changyonghu : Were you able to test that this works.
> 
> LGTM other than minor comment
Yes, I am able to manually test. ./scripts/cti --pr 4253 --health-check --duration 10 runs 10 sec after setup.

**Comment by ankushagarwal** [comment]
@changyonghu you can merge using `@bors-libra r=ankushagarwal` command when you're ready

**Comment by andll** [comment]
@bors-libra r-
Looks like bors read the message in quotes, should probably address comment before merging

File: testsuite/cluster-test/src/main.rs [hunk 2]
```diff
@@ -155,7 +155,11 @@ pub fn main() {
     let mut perf_msg = None;
 
     if args.health_check {
-        runner.run_health_check();
+        let duration = Duration::from_secs(args.duration);
+        if runner.run_health_check(duration).is_err() {
```
Comment (andll) [inline testsuite/cluster-test/src/main.rs:159]: You can use `exit_on_error(runner.run_health_check(duration))`

File: testsuite/cluster-test/src/main.rs (hunk 1 above)
Comment (andll) [reply testsuite/cluster-test/src/main.rs:758]: Also I think return here will be more obvious/appropriate then break

File: testsuite/cluster-test/src/main.rs [hunk 3]
```diff
@@ -155,7 +155,11 @@ pub fn main() {
     let mut perf_msg = None;
 
     if args.health_check {
-        runner.run_health_check();
+        let duration = Duration::from_secs(args.duration);
```
Comment (snakech) [inline testsuite/cluster-test/src/main.rs:158]: @andll @ankushagarwal is accuracy of duration unit by secs enough?

Comment (andll) [reply testsuite/cluster-test/src/main.rs:158]: Yes

**Pull request closed without merging**